test/jdk/java/lang/invoke/VarHandles/LazyStaticTest.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8291065
+ * @summary Ensures creation of static field VarHandle does not trigger
+ *          class initialization.
+ * @run main LazyStaticTest
+ */
+
+import java.lang.invoke.MethodHandles;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class LazyStaticTest {
+    static Set<Class<?>> initialized = ConcurrentHashMap.newKeySet();
+
+    public static void main(String... args) throws Throwable {
+        assert initialized.isEmpty() : "Incorrect initial state";
+
+        class Sample1 {
```

Comment:
Nit: suggest to move this inner class outside of the method body.  I think it helps readability.   maybe name it as `Sample` or `Test` or sth.   I expect `Sample2` when I read `Sample1`.